### PR TITLE
feat(wordcloud): click and over events on text

### PR DESCRIPTION
**Pull request by markov00**
## Summary

The `<Worldcloud>` chart now fully support the `onElementClick`, `onElementOver` and `onElementOut` events, returning the original datum the callback.

## Details

The following fixes/improvements are applied:
- fix the `wordcloud` API to use the correct `Config` type (previously the `PartitionConfig` was listed)
- removed the event `callers` from the wordcloud state (no more required as we can call the actions directly from the SVG `text` element)
- the input `datum` is added to the dataset passed to the `d3-cloud` function. This because the dataset is mutated by `d3-cloud` causing the `WordModel` prop `weight` to be overwritten by the `fontWeight` value.
 
## Connected issues

This closes the support request for the wordcloud migration in Kibana https://github.com/elastic/elastic-charts/issues/1156

### Checklist

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials

### CODE REVIEW

**Review by rshen91 — APPROVED** [review]
LGTM tested locally

**Review by monfera — APPROVED** [review]
Looks good, tested all three event types on all the sample configs, and the code improvements are good too!

File: src/chart_types/wordcloud/renderer/svg/connected_component.tsx [hunk 1]
```diff
@@ -125,36 +128,74 @@ function layoutMaker(config: Configs, data: Datum[]) {
...
+            ...(onElementOver && {
+              onMouseOver: () => {
+                onElementOver?.([[datum, { specId, key: specId }]]);
+              },
+            }),
+            ...(onElementOut && {
+              onMouseOut: () => {
+                onElementOut?.();
+              },
+            }),
```
Comment (nickofthyme) [inline src/chart_types/wordcloud/renderer/svg/connected_component.tsx:166]: Unneeded optional chaining

```suggestion
              onClick: () => {
                onElementClick([[datum, { specId, key: specId }]]);
              },
            }),
            ...(onElementOver && {
              onMouseOver: () => {
                onElementOver([[datum, { specId, key: specId }]]);
              },
            }),
            ...(onElementOut && {
              onMouseOut: () => {
                onElementOut();
              },
            }),
```

File: src/chart_types/wordcloud/specs/index.ts [hunk 2]
```diff
@@ -24,9 +24,14 @@ import { Spec } from '../../../specs';
...
 import { RecursivePartial } from '../../../utils/common';
-import { Config } from '../../partition_chart/layout/types/config_types';
 import { config } from '../layout/config/config';
-import { WordModel, defaultWordcloudSpec, WeightFn, OutOfRoomCallback } from '../layout/types/viewmodel_types';
+import {
+  WordModel,
+  defaultWordcloudSpec,
+  WeightFn,
+  OutOfRoomCallback,
+  Configs as WordcloudConfigs,
```
Comment (nickofthyme) [inline src/chart_types/wordcloud/specs/index.ts:33]: I think all the `Config` types should be prefixed, I think there are three `Config` types across charts that are all different. Not for this pr

**Review by nickofthyme — APPROVED** [review]
Great work! Changes look good to me, left one code change. Tested locally and functions as expected.

File: src/chart_types/wordcloud/specs/index.ts (hunk 2 above)
Comment (markov00) [reply src/chart_types/wordcloud/specs/index.ts:33]: Yes absolutely agree with you

File: src/chart_types/wordcloud/renderer/svg/connected_component.tsx (hunk 1 above)
Comment (markov00) [reply src/chart_types/wordcloud/renderer/svg/connected_component.tsx:166]: 🤦

Comment (markov00) [reply src/chart_types/wordcloud/renderer/svg/connected_component.tsx]: fixed here https://github.com/elastic/elastic-charts/pull/1180/commits/1e8fb67e5bb73d1ae81e69165085e94c52044cbe

File: src/chart_types/wordcloud/specs/index.ts (hunk 2 above)
Comment (markov00) [reply src/chart_types/wordcloud/specs/index.ts:33]: I opened an issue here to track this: https://github.com/elastic/elastic-charts/issues/1189

**Pull request merged**